src/app-layout/__tests__/multi-layout.test.tsx

```diff
@@ -75,6 +75,25 @@ describeEachAppLayout({ themes: ['refresh-toolbar'], sizes: ['desktop'] }, () =>
     expect(isDrawerClosed(firstLayout.findNavigation())).toEqual(false);
   });
 
+  test('merges navigation from two instances with navigation hidden in primary', async () => {
+    const { firstLayout, secondLayout } = await renderAsync(
+      <AppLayout
+        {...defaultAppLayoutProps}
+        data-testid="first"
+        navigation="testing nav"
+        navigationHide={true}
```

Comment:
The second layout using navigationHide is a more common case.

Reply:
I updated the test name and added some assertions below to cover this case more thoroughly.